booster/library/Booster/SMT/Interface.hs

```diff
@@ -302,8 +299,9 @@ checkPredicates ctxt givenPs givenSubst psToCheck
         consistent <- smtRun CheckSat
         when (consistent /= Sat) $ do
             void $ smtRun Pop
-            Log.logMessage ("Inconsistent ground truth, check returns Nothing" :: Text)
-            fail "returns nothing"
+            let errMsg = ("Inconsistent ground truth, check returns Nothing" :: Text)
+            Log.logMessage errMsg
+            throwE $ GeneralSMTError errMsg
```

Comment:
This is not really an "error", rather a defensive strategy to avoid jumping to unjustified conclusions